pkg/k8s/portforward.go

```diff
@@ -124,7 +124,14 @@ func newPortForward(
 	emitLogs bool,
 ) (*PortForward, error) {
 
-	req := k8sAPI.CoreV1().RESTClient().Post().
+	restClient := k8sAPI.CoreV1().RESTClient()
+	if fakeRest, ok := restClient.(*rest.RESTClient); ok {
+		if fakeRest == nil {
+			return nil, nil
+		}
+	}
```

Comment:
if ok == false, i assume something else is broken, perhaps we return an error in that case?

Reply:
If ok == false, it just means that the internal implementation of the interface has changed which shouldn't be a problem when this is called in normal operation (not in a test).  I tried to make this as conservative as possible: i.e. only return early if we can be absolutely sure we're in a test.  Added a comment to give more context to this.